polaris-core/src/main/java/org/apache/polaris/core/persistence/BasePersistence.java

```diff
@@ -0,0 +1,341 @@
+    PolarisBaseEntity baseEntity = lookupEntityByName(callCtx, catalogId, parentId, typeCode, name);
+    if (baseEntity == null) {
+      return null;
+    }
+    return new PolarisEntityActiveRecord(baseEntity);
+  }
+
+  /**
+   * Lookup a set of entities given their catalog id/entity id unique identifier
+   *
+   * @param callCtx call context
+   * @param entityIds list of entity ids
+   * @return list of polaris base entities, parallel to the input list of ids. An entity in the list
+   *     will be null if the corresponding entity could not be found.
+   */
+  @Nonnull
+  List<PolarisBaseEntity> lookupEntities(
+      @Nonnull PolarisCallContext callCtx, List<PolarisEntityId> entityIds);
+
+  /**
+   * Lookup the current entityVersion of an entity given its catalog id (which can be NULL_ID for
+   * top-level entities) and its unique id. Will return 0 if the entity does not exist.
+   *
+   * @param callCtx call context
+   * @param catalogId catalog id or NULL_ID
+   * @param entityId unique entity id
+   * @return current version for that entity or 0 if entity was not found.
+   */
+  int lookupEntityVersion(@Nonnull PolarisCallContext callCtx, long catalogId, long entityId);
+
+  /**
+   * Get change tracking versions for all specified entity ids.
+   *
+   * @param callCtx call context
+   * @param entityIds list of entity id
+   * @return list parallel to the input list of entity versions. If an entity cannot be found, the
+   *     corresponding element in the list will be null
+   */
+  @Nonnull
+  List<PolarisChangeTrackingVersions> lookupEntityVersions(
+      @Nonnull PolarisCallContext callCtx, List<PolarisEntityId> entityIds);
+
+  /**
+   * List all active entities of the specified type which are child entities of the specified parent
+   *
+   * @param callCtx call context
+   * @param catalogId catalog id for that entity, NULL_ID if the entity is top-level
+   * @param parentId id of the parent, can be the special 0 value representing the root entity
+   * @param entityType type of entities to list
+   * @return the list of entities_active records for the specified list operation
+   */
+  @Nonnull
+  List<PolarisEntityActiveRecord> listActiveEntities(
+      @Nonnull PolarisCallContext callCtx,
+      long catalogId,
+      long parentId,
+      @Nonnull PolarisEntityType entityType);
+
+  /**
+   * List active entities where some predicate returns true
+   *
+   * @param callCtx call context
+   * @param catalogId catalog id for that entity, NULL_ID if the entity is top-level
+   * @param parentId id of the parent, can be the special 0 value representing the root entity
+   * @param entityType type of entities to list
+   * @param entityFilter the filter to be applied to each entity. Only entities where the predicate
+   *     returns true are returned in the list
+   * @return the list of entities for which the predicate returns true
+   */
+  @Nonnull
+  List<PolarisEntityActiveRecord> listActiveEntities(
```

Comment:
suggestion: `PolarisEntityActiveRecord` -> `ShallowEntityRecord`... "shallow" to indication it contains only IDs, but not the "body".

Reply:
Ah didn't see this until I already renamed to `EntityNameLookupRecord`. I had considered something similar to `ShallowEntityRecord` but then we already have `PolarisEntityCore` which is a different subset of `PolarisBaseEntity` so that might be confusing.

For now, the usage of this class happens to reflect the idea of a NameLookup, so that might be more clear. I guess I could also call it `ShallowNameLookupRecord` in case people are confused why a name lookup doesn't include the whole entity.

Reply:
`EntityNameLookupRecord` SGTM :+1: